[topgen] Adding an OTP clock connection to OTBN

I'm having trouble adding an OTP clock connection to OTBN, I altered `top_earlgrey.hjson` to add the connection:

```
    { name: "otbn",
      type: "otbn",
      clock_srcs: {clk_i: "main", clk_edn_i: "main", clk_otp_i: "io_div4"},
      clock_group: "trans",
      reset_connections: {rst_ni: "sys", rst_edn_ni: "sys", rst_otp_ni: "lc_io_div4"},
      base_addr: "0x411D0000",
    },
```

And made the relevant changes to `otbn.hjson`

```
  clock_primary: "clk_i"
  other_clock_list: [../sw/device/lib/dif/dif_clkmgr_unittest.cc:19:3: error: ‘CLKMGR_CLK_HINTS_STATUS_CLK_MAIN_OTBN_VAL_BIT’ was
not declared in this scope
   CLKMGR_CLK_HINTS_STATUS_CLK_MAIN_OTBN_VAL_BIT

    "clk_edn_i",
    "clk_otp_i"
  ],
  reset_primary: "rst_ni",
  other_reset_list: [
    "rst_edn_ni",
    "rst_otp_ni"
  ],
```

Then topgen is run

```
util/topgen.py -t hw/top_earlgrey/data/top_earlgrey.hjson -o hw/top_earlgrey/ --rnd_cnst_seed 4881560218908238235
```

Then I build the earlgrey verilator simulation, resulting in the following error:

```%Error: ../src/lowrisc_systems_clkmgr_pkg_0.1/rtl/autogen/clkmgr_pkg.sv:20:5: Duplicate declaration of enum value: Otbn
   20 |     Otbn = 4
      |     ^~~~
        ../src/lowrisc_systems_clkmgr_pkg_0.1/rtl/autogen/clkmgr_pkg.sv:19:5: ... Location of original declaration
   19 |     Otbn = 3,
      |     ^~~~
%Error: Exiting due to 1 error(s)
```

I noticed other IPs using the OTP clock had a `clock_group` set to `"secure"` so I tried that. This fixed the verilator build error but breaks the software build:


```
../sw/device/lib/dif/dif_clkmgr_unittest.cc:19:3: error: ‘CLKMGR_CLK_HINTS_STATUS_CLK_MAIN_OTBN_VAL_BIT’ was
not declared in this scope
   CLKMGR_CLK_HINTS_STATUS_CLK_MAIN_OTBN_VAL_BIT
```

Any idea what's going wrong?

--- a/hw/ip/clkmgr/data/clkmgr_pkg.sv.tpl
+++ b/hw/ip/clkmgr/data/clkmgr_pkg.sv.tpl
@@ -13,8 +13,8 @@ num_hints = len(hint_clks)
 package clkmgr_pkg;
 
   typedef enum int {
-% for hint, v in hint_clks.items():
-    ${v['name'].capitalize()} = ${loop.index}${"," if not loop.last else ""}
+% for idx, blk_name in list(enumerate(hint_blocks)):
+    ${blk_name.capitalize()} = ${idx}${"," if not loop.last else ""}
 % endfor
   } hint_names_e;
 

--- a/util/topgen.py
+++ b/util/topgen.py
@@ -483,6 +483,10 @@ def generate_clkmgr(top, cfg_path, out_path):
     hint_clks = {clk: {'name': clk.rsplit('_', 1)[-1], 'src': src}
                  for clk, src in hints.items()}
 
+    # The names of blocks that use one or more sw hint clocks (clkmgr has an
+    # "idle" feedback signal from each), in ascending order.
+    hint_blocks = sorted(set([v['name'] for v in hint_clks.values()]))
+
     # Define a canonical ordering for rg_srcs
     rg_srcs = sorted(rg_srcs_set)
 
@@ -498,7 +502,8 @@ def generate_clkmgr(top, cfg_path, out_path):
                                  rg_clks=rg_clks,
                                  sw_clks=sw_clks,
                                  export_clks=top['exported_clks'],
-                                 hint_clks=hint_clks)
+                                 hint_clks=hint_clks,
+                                 hint_blocks=hint_blocks)
             except:  # noqa: E722
                 log.error(exceptions.text_error_template().render())
 

